### PR TITLE
MM-57699 Looker changes for Desktop/Webapp split to fct_active_users

### PR DESCRIPTION
**Pull request by sri-byte**
Impact: These are the looker changes for MM-57699. 

Please only merge after this PR is merged. https://github.com/mattermost/mattermost-data-warehouse/pull/1478

### CODE REVIEW

File: views/marts/product/fct_active_users.view.lkml [hunk 1]
```diff
@@ -137,34 +193,90 @@ view: fct_active_users {
...
-    label: "Total Legacy Desktop/Client WAU"
+    sql: ${weekly_legacy_webapp_active_users} ;;
+    label: "Total Legacy Webapp WAU"
     view_label: "  * Metrics. WAU"
   }
 
   dimension: weekly_mobile_active_users {
     type: number
-    sql: ${TABLE}.weekly_mobile_active_users ;;
+    sql: ${TABLE}.monthly_mobile_active_users ;;
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:279]: monthly_mobile_active_users -> weekly_mobile_active_users

File: views/marts/product/fct_active_users.view.lkml [hunk 2]
```diff
@@ -76,28 +76,84 @@ view: fct_active_users {
...
+    type: sum
+    sql: ${daily_client_desktop_active_users} ;;
+    label: "Total Client Desktop DAU"
+    view_label: "  * Metrics. DAU"
+  }
+
+  dimension: daily_legacy_desktop_active_users {
+    type: number
+    sql: ${TABLE}.daily_legacy_desktop_active_users ;;
+    label: "Legacy Desktop DAU"
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:121]: Why not "Client Legacy Desktop DAU"? (or remove "Client" from "Client Desktop DAU" above)?

File: views/marts/product/fct_active_users.view.lkml [hunk 3]
```diff
@@ -137,34 +193,90 @@ view: fct_active_users {
...
     type: sum
     sql: ${weekly_desktop_active_users} ;;
-    label: "Total Desktop/Client WAU"
+    label: "Total Desktop WAU"
+    view_label: "  * Metrics. WAU"
+  }
+
+  dimension: weekly_webapp_active_users {
+    type: number
+    sql: ${TABLE}.weekly_webapp_active_users ;;
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:209]: to be consistent with above, why not `weekly_client_webapp_active_users` ; or to simplify, remove client from all?

File: views/marts/product/fct_active_users.view.lkml [hunk 4]
```diff
@@ -76,28 +76,84 @@ view: fct_active_users {
...
+  }
+
+  dimension: daily_client_desktop_active_users {
+    type: number
+    sql: ${TABLE}.daily_client_desktop_active_users ;;
+    label: "Client Desktop DAU"
+    description: "Client DAU reported by Rudderstack telemetry."
+  }
+
+  measure: total_client_daily_desktop_active_users {
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:111]: `total_client_daily_desktop_active_users` -> `total_daily_client_desktop_active_users` to be consistent with below?

File: views/marts/product/fct_active_users.view.lkml [hunk 5]
```diff
@@ -137,34 +193,90 @@ view: fct_active_users {
...
+  }
+
+  dimension: weekly_client_desktop_active_users {
+    type: number
+    sql: ${TABLE}.weekly_client_desktop_active_users ;;
+    label: "Client Desktop WAU"
+    description: "Client WAU reported by Rudderstack telemetry."
+  }
+
+  measure: total_client_weekly_desktop_active_users {
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:228]: `total_client_weekly_desktop_active_users` -> `total_weekly_client_desktop_active_users` to be consistent with the sql query below?

File: views/marts/product/fct_active_users.view.lkml (hunk 1 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: this is updated, mobile as a whole has been dropped from the dbt model, so this has been removed.

File: views/marts/product/fct_active_users.view.lkml [hunk 6]
```diff
@@ -211,31 +323,87 @@ view: fct_active_users {
...
-    label: "Total Legacy Desktop/Client MAU"
+    sql: ${monthly_webapp_active_users} ;;
+    label: "Total Webapp MAU"
     view_label: "  * Metrics. MAU"
   }
 
+  dimension: monthly_client_desktop_active_users {
+      type: number
+      sql: ${TABLE}.monthly_client_desktop_active_users ;;
+      label: "Client Webapp MAU"
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: Fixed now.

File: views/marts/product/fct_active_users.view.lkml (hunk 5 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: Fixed.

File: views/marts/product/fct_active_users.view.lkml [hunk 7]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
   dimension: daily_desktop_active_users {
     type: number
     sql: ${TABLE}.daily_desktop_active_users ;;
-    label: "Desktop/Client DAU"
-    description: "DAU reported by Rudderstack telemetry."
+    label: "Desktop DAU"
+    description: "Desktop DAU reported by telemetry."
```
Comment (ifoukarakis) [inline views/marts/product/fct_active_users.view.lkml:80]: It's worth mentioning that this is coming from client. The same convention as in DAU can be used for label and description.

File: views/marts/product/fct_active_users.view.lkml [hunk 8]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
...
 
-  dimension: daily_legacy_active_users {
+  dimension: daily_webapp_active_users {
     type: number
-    sql: ${TABLE}.daily_legacy_active_users ;;
-    label: "Legacy Desktop/Client DAU"
-    description: "DAU reported by Segment telemetry."
+    sql: ${TABLE}.daily_webapp_active_users ;;
+    label: "Webapp DAU"
+    description: "Webapp DAU reported by telemetry."
```
Comment (ifoukarakis) [inline views/marts/product/fct_active_users.view.lkml:94]: Same here. Please mention it's client DAU.

File: views/marts/product/fct_active_users.view.lkml [hunk 9]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
...
+
+  measure: total_daily_webapp_active_users {
+    type: sum
+    sql: ${daily_webapp_active_users} ;;
+    label: "Total Webapp DAU"
+    view_label: "  * Metrics. DAU"
   }
 
-  measure: total_daily_legacy_active_users {
+  dimension: daily_client_desktop_active_users {
```
Comment (ifoukarakis) [inline views/marts/product/fct_active_users.view.lkml:104]: Is this different than `daily_desktop_active_users`? If not, makes sense to keep only one of the two.

File: views/marts/product/fct_active_users.view.lkml [hunk 10]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
...
+  }
+
+  measure: total_daily_legacy_desktop_active_users {
+    type: sum
+    sql: ${daily_legacy_desktop_active_users} ;;
+    label: "Total Legacy Desktop DAU"
+    view_label: "  * Metrics. DAU"
+  }
+
+  dimension: daily_client_webapp_active_users {
```
Comment (ifoukarakis) [inline views/marts/product/fct_active_users.view.lkml:132]: Is this different from `daily_webapp_active_users`? If yes, label should be explicit. If not, only one should be kept.

File: views/marts/product/fct_active_users.view.lkml [hunk 11]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
...
+    label: "Total Webapp DAU"
+    view_label: "  * Metrics. DAU"
   }
 
-  measure: total_daily_legacy_active_users {
+  dimension: daily_client_desktop_active_users {
+    type: number
+    sql: ${TABLE}.daily_client_desktop_active_users ;;
+    label: "Client Desktop DAU"
+    description: "Client DAU reported by Rudderstack telemetry."
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:108]: "Client" -> "Client Desktop"?

File: views/marts/product/fct_active_users.view.lkml [hunk 12]
```diff
@@ -76,31 +76,86 @@ view: fct_active_users {
...
+    sql: ${daily_legacy_desktop_active_users} ;;
+    label: "Total Legacy Desktop DAU"
+    view_label: "  * Metrics. DAU"
+  }
+
+  dimension: daily_client_webapp_active_users {
+    type: number
+    sql: ${TABLE}.daily_client_webapp_active_users ;;
+    label: "Client Webapp DAU"
+    description: "Webapp DAU reported by Rudderstack telemetry."
```
Comment (ifoukarakis) [inline views/marts/product/fct_active_users.view.lkml:136]: Description mentions Rudderstack, while label doesn't distinguish.

**Review by ifoukarakis — COMMENTED** [review]
Left a few comments related to naming and possibly duplicate metrics. The comments are for daily, but the same apply for weekly and monthly.

File: views/marts/product/fct_active_users.view.lkml [hunk 13]
```diff
@@ -123,28 +178,84 @@ view: fct_active_users {
...
+    sql: ${weekly_webapp_active_users} ;;
+    label: "Total Webapp WAU"
+    view_label: "  * Metrics. WAU"
+  }
+
+  dimension: weekly_client_desktop_active_users {
+    type: number
+    sql: ${TABLE}.weekly_client_desktop_active_users ;;
+    label: "Client Desktop WAU"
+    description: "Client WAU reported by Rudderstack telemetry."
```
Comment (catalintomai) [inline views/marts/product/fct_active_users.view.lkml:210]: "Client" -> "Client Desktop" or "Desktop"?

File: views/marts/product/fct_active_users.view.lkml (hunk 11 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: Updated to `Client Desktop`

File: views/marts/product/fct_active_users.view.lkml (hunk 7 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: This is both Client (Rudderstack) and Legacy (Segment). I updated it to reflect the same.

File: views/marts/product/fct_active_users.view.lkml (hunk 8 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: Updated to client & legacy, which are the sources this dimension is made of

File: views/marts/product/fct_active_users.view.lkml (hunk 9 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml:104]: this is different, we have 2 cohorts split by 2 sources.
`client_desktop` -> Rudderstack and Desktop
`legacy_desktop` -> Segment and Desktop
`client_webapp` -> Rudderstack and Webapp
`legacy_webapp` -> Segment and Webapp
and
`desktop_active_users` -> Rudderstack & Segment Desktop
`webapp_active_users` -> Rudderstack & Segment Webapp
and
`daily_active_users` -> Rudderstack & Segment Desktop & Webapp

File: views/marts/product/fct_active_users.view.lkml (hunk 10 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml:132]: Added all the possible combinations to the above comment.

File: views/marts/product/fct_active_users.view.lkml (hunk 13 above)
Comment (sri-byte) [reply views/marts/product/fct_active_users.view.lkml]: This is now updated.

**Review by catalintomai — APPROVED** [review]

**Review by ifoukarakis — APPROVED** [review]

**Pull request merged**